http-common/src/server.rs

```diff
@@ -340,31 +326,54 @@ impl Error {
         let body = crate::ErrorBody {
             message: std::borrow::Cow::Borrowed(std::borrow::Borrow::borrow(&self.message)),
         };
-        let res = json_response(self.status_code, Some(&body));
+        let res = response::json(self.status_code, &body);
         res
     }
 }
 
 #[cfg(feature = "tokio1")]
-pub fn json_response(
-    status_code: hyper::StatusCode,
-    body: Option<&impl serde::Serialize>,
-) -> hyper::Response<hyper::Body> {
-    let res = hyper::Response::builder().status(status_code);
-    // `res` is consumed by both branches, so this cannot be replaced with `Option::map_or_else`
-    //
-    // Ref: https://github.com/rust-lang/rust-clippy/issues/5822
-    #[allow(clippy::option_if_let_else)]
-    let res = if let Some(body) = body {
+pub mod response {
+    pub fn empty() -> hyper::Response<hyper::Body> {
+        let res = hyper::Response::builder()
+            .status(hyper::StatusCode::NO_CONTENT)
+            .body(Default::default())
+            .expect("cannot fail to build hyper response");
+
+        res
+    }
+
+    pub fn json(
+        status_code: hyper::StatusCode,
+        body: &impl serde::Serialize,
+    ) -> hyper::Response<hyper::Body> {
         let body = serde_json::to_string(body).expect("cannot fail to serialize response to JSON");
         let body = hyper::Body::from(body);
-        res.header(hyper::header::CONTENT_TYPE, "application/json")
-            .body(body)
-    } else {
-        res.body(Default::default())
-    };
-    let res = res.expect("cannot fail to build hyper response");
-    res
+
+        let res = hyper::Response::builder()
+            .status(status_code)
+            .header(hyper::header::CONTENT_TYPE, "application/json")
+            .body(body);
+
+        let res = res.expect("cannot fail to build hyper response");
+        res
+    }
+
+    pub fn zip(
+        status_code: hyper::StatusCode,
+        size: usize,
+        body: Vec<u8>,
+    ) -> hyper::Response<hyper::Body> {
+        let res = hyper::Response::builder().status(status_code);
+
+        let res = res
+            .header(hyper::header::CONTENT_ENCODING, "zip")
```

Comment:
This header is probably wrong, but it's what the current edged uses. So I left it.

Reply:
Yes, it's wrong. It should be `deflate`.

It should be fine to change it - it was already a bad value so no self-respecting client library would be depending on it anyway.

Reply:
Ok, fixed it.